src/continuous-validation.cc

```diff
@@ -0,0 +1,232 @@
+//
+// Copyright (c) 2014, 2015, 2016, 2017, 2018 CNRS
+// Authors: Florent Lamiraux, Joseph Mirabel, Diane Bury
+//
+// This file is part of hpp-core
+// hpp-core is free software: you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version
+// 3 of the License, or (at your option) any later version.
+//
+// hpp-core is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+// of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Lesser Public License for more details.  You should have
+// received a copy of the GNU Lesser General Public License along with
+// hpp-core  If not, see
+// <http://www.gnu.org/licenses/>.
+
+#include <hpp/core/continuous-validation.hh>
+
+#include <limits>
+#include <pinocchio/multibody/geometry.hpp>
+#include <hpp/util/debug.hh>
+#include <hpp/core/collision-path-validation-report.hh>
+#include <hpp/core/straight-path.hh>
+#include <hpp/core/path-vector.hh>
+#include <hpp/core/continuous-validation/initializer.hh>
+#include <hpp/core/continuous-validation/solid-solid-collision.hh>
+
+#include <iterator>
+namespace hpp {
+  namespace core {
+    /// Validate interval centered on a path parameter
+    /// \param config Configuration at abscissa tmin on the path.
+    /// \param tmin parameter value in the path interval of definition
+    /// \retval tmin lower bound of valid interval if reverve is true,
+    ///              upper bound of valid interval if reverse is false.
+    ///              other bound of valid interval is input tmin.
+    /// \return true if the configuration is collision free for this parameter
+    ///         value, false otherwise.
+    bool ContinuousValidation::validateConfiguration(const Configuration_t &config, const value_type &t,
+                                                            interval_t &interval, PathValidationReportPtr_t &report)
+    {
+      interval.first = -std::numeric_limits <value_type>::infinity ();
+      interval.second = std::numeric_limits <value_type>::infinity ();
+      interval_t tmpInt;
+      robot_->currentConfiguration (config);
+      robot_->computeForwardKinematics();
+      robot_->updateGeometryPlacements();
+      BodyPairCollisions_t::iterator itMin = bodyPairCollisions_.begin();
+      value_type distance = 0;
+      if (!validateIntervals<BodyPairCollisions_t, CollisionValidationReportPtr_t>
+            (bodyPairCollisions_, t, interval, tmpInt, report, distance))
+        return false;
+      std::advance(itMin, distance);
+      if (bodyPairCollisions_.size() > 1 && itMin != bodyPairCollisions_.begin())
+        std::iter_swap (bodyPairCollisions_.begin(), itMin);
```

Comment:
What is the intend of this swap ?

Reply:
I think I am the one who wrote this line. I will add a comment. Sorry for the noise.

Reply:
I did not add it, but I think the swap is meant to put the most constraining BodyPairCollision at the beginning of the vector to try and optimize the next iteration in case the most constraining BodyPairCollision is the same, so that the tested interval for all the following pairs is smaller than if it was another BodyPairCollision at the beginning of the vector